smart-home-application/src/Room.cpp

```diff
@@ -4,6 +4,55 @@
 #include <algorithm>
 
 namespace smart_home_app {
+    Room::Room(std::string room_name) : room_name_(room_name) {};
 
+    Room::~Room() {
+        for (Device* d : devices_) {
+            delete d;
+        }
+        devices_.clear();
+    };
+
+    Room::Room(const Room& other) : room_name_(other.room_name_), devices_(other.devices_) {};
+
+    Room& Room::operator=(const Room& other) {
+        Room tmp(other);
+        std::swap(room_name_, tmp.room_name_);
+        std::swap(devices_, tmp.devices_);
+        return *this;
+    };
+    
+    Room::Room(Room&& other) : room_name_(std::move(other.room_name_)), devices_(std::move(other.devices_)) {};
+
+    Room& Room::operator=(Room&& other) {
+        room_name_ = std::move(other.room_name_);
```

Comment:
incorrect implementation of move assignment operator, need to check first if other is different than *this